plugins/identity/app/services/service_layer/identity_services/credential.rb

```diff
@@ -2,19 +2,52 @@
 
 module ServiceLayer
   module IdentityServices
-    # This module implements Openstack User API
+    # This module implements Openstack Credential API
     module Credential
 
       AWSCreds = Struct.new(:access, :secret, :error)
 
-      def aws_credentials(user_id, filter = {})
+      def aws_credentials(user_id, project_id)
+        aws_creds = [] 
         response = elektron_identity.get("users/#{user_id}/credentials/OS-EC2")
         if !response.body["credentials"].empty?
-          creds_hash = response.body["credentials"].first
-          aws_creds = AWSCreds.new(creds_hash["access"] , creds_hash["secret"], "")
+          creds_hash = response.body["credentials"]
+          creds_hash.each do | creds |
+            next if creds['tenant_id'] != project_id
+            aws_creds << AWSCreds.new(creds["access"] , creds["secret"], "") 
+          end
         else
           err = AWSCreds.new("" , "", "AWS EC2 credentials are not created. Without this, email service will not work. Open your web-console and execute `openstack ec2 credentials create` command")
         end
+        err ? err : aws_creds.first
+      end
+
+      def all_ec2_creds(user_id, project_id)
+        creds = []
+        err = ""
+        if user_id && project_id
+          res = elektron_identity.get("users/#{user_id}/credentials/OS-EC2")
+          if !res.body["credentials"].empty?
+            creds_hash = response.body["credentials"]
+            creds_hash.each do | creds |
+              next if creds['tenant_id'] != project_id
+              aws_creds << AWSCreds.new(creds["access"] , creds["secret"], "") 
+          end
+          else
+            err = AWSCreds.new("" , "", "AWS EC2 credentials are not created. Without this, email service will not work. Open your web-console and execute `openstack ec2 credentials create` command")
+          end
+        else
+          err = "user_id / project can't be empty"
+        end
+        [err, creds]
+      end
+
+      def get_project(project_id, user_id)
+        res = elektron_identity.get("projects/#{project_id}/users/#{user_id}")
+      end
+
+      def get_user(user_id)
```

Comment:
Please use the corresponding service layer method. Please do not recreate methods from other service layers, this could lead to errors. Use the ones defined here:
https://github.com/sapcc/elektra/blob/master/plugins/identity/app/services/service_layer/identity_services/user.rb

Reply:
Those were removed in the subsequent commit.